scripts/kconfig/kconfigfunctions.py

```diff
@@ -91,8 +104,172 @@ def dt_str_val(kconf, _, name):
+        None        No division
+        'k' or 'K'  divide by 1024 (1 << 10)
+        'm' or 'M'  divide by 1,048,576 (1 << 20)
+        'g' or 'G'  divide by 1,073,741,824 (1 << 30)
+    """
+
+    if doc_mode or edt is None:
+        return "0x0"
+
+    dev = edt.chosen_dev(chosen)
+
+    return _dt_to_reg_addr(dev, index, unit)
+
+
+def dt_chosen_to_reg_size(kconf, _, chosen, index=0, unit=None):
+    """
+    This function takes a 'chosen' property and treats that property as a path
+    to a EDT device.  If it finds a EDT device, it will look to see if that
+    device has a register at the give 'index' and return the size value of
+    that reg, if not we return 0.
+
+    The function will divide the value based on 'unit':
+        None        No division
+        'k' or 'K'  divide by 1024 (1 << 10)
+        'm' or 'M'  divide by 1,048,576 (1 << 20)
+        'g' or 'G'  divide by 1,073,741,824 (1 << 30)
+    """
+
+    if doc_mode or edt is None:
+        return "0"
+
+    dev = edt.chosen_dev(chosen)
+
+    return _dt_to_reg_size(dev, index, unit)
+
+
+def dt_path_to_reg_addr(kconf, _, path, index=0, unit=None):
+    """
+    This function takes a 'path' and looks for a EDT device at that path.
+    If it finds a EDT device, it will look to see if that device has a
+    register at the give 'index' and return the address value of that reg, if
+    not we return 0.
+
+    The function will divide the value based on 'unit':
+        None        No division
+        'k' or 'K'  divide by 1024 (1 << 10)
+        'm' or 'M'  divide by 1,048,576 (1 << 20)
+        'g' or 'G'  divide by 1,073,741,824 (1 << 30)
+    """
+
+    if doc_mode or edt is None:
+        return "0"
+
+    try:
+       dev = edt.get_dev(path)
+    except edtlib.EDTError:
+       return "0"
+
+    return _dt_to_reg_addr(dev, index, unit)
+
+
+def dt_path_to_reg_size(kconf, _, path, index=0, unit=None):
+    """
+    This function takes a 'path' and looks for a EDT device at that path.
+    If it finds a EDT device, it will look to see if that device has a
+    register at the give 'index' and return the size value of that reg, if
+    not we return 0.
+
+    The function will divide the value based on 'unit':
+        None        No division
+        'k' or 'K'  divide by 1024 (1 << 10)
+        'm' or 'M'  divide by 1,048,576 (1 << 20)
+        'g' or 'G'  divide by 1,073,741,824 (1 << 30)
+    """
+
+    if doc_mode or edt is None:
+        return "0"
+
+    try:
+        dev = edt.get_dev(path)
+    except edtlib.EDTError:
+       return "0"
+
+    return _dt_to_reg_size(dev, index, unit)
+
+
+def dt_compat_enabled(kconf, _, compat):
+    """
+    This function takes a 'compat' and returns "y" if we find an "enabled"
+    compatible device in the EDT otherwise we return "n"
+    """
+
+    if doc_mode or edt is None:
+        return "n"
+
+    compats = edt.compat_devs(compat)
+    for dev in compats:
+        if dev.enabled:
+            return "y"
+
+    return "n"
```

Comment:
Could remove `compat_devs()` and do

```python
for dev in edt.devices:
    if compat in dev.compats and dev.enabled:
        return "y"

return "n"
```

It'll be fast enough.

Reply:
Done.